File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        c={}
        left =0
        max_len =0
        
        for i in range(len(s)):
            c[s[i]]= 1+c.get(s[i],0)
            
            while(((i-left+1)-max(c.values())>k)):
                
                c[s[left]]-=1
                left+=1

            max_len = max(max_len,(i-left+1))


        return max_len




            

             




        

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.cpp
class Solution {
public:
    
    int diameterOfBinaryTree(TreeNode* root) {
        int max_val =0;
        dfs(root,max_val);
        return max_val;
    }
private:
    int dfs(TreeNode*root, int&max_val){
        if(!root){
            return 0;
        }
        int left = dfs(root->left, max_val);
        int right = dfs(root->right, max_val);
        max_val = max(max_val ,left+right);
        return 1+max(left, right);
    }
};
